lib/ansible/modules/cloud/cloudstack/cs_physical_network.py

```diff
@@ -0,0 +1,470 @@
+      - VirtualRouter
+      - InternalLbVm
+      - VpcVirtualRouter
+
+# Ensure a network is disabled
+- local_action:
+    module: cs_physical_network
+    name: net01
+    zone: zone01
+    state: disabled
+
+# Ensure a network is enabled
+- local_action:
+    module: cs_physical_network
+    name: net01
+    zone: zone01
+    state: enabled
+
+# Ensure a network is absent
+- local_action:
+    module: cs_physical_network
+    name: net01
+    zone: zone01
+    state: absent
+'''
+
+RETURN = '''
+---
+id:
+  description: UUID of the network.
+  returned: success
+  type: str
+  sample: 3f8f25cd-c498-443f-9058-438cfbcbff50
+name:
+  description: Name of the network.
+  returned: success
+  type: str
+  sample: net01
+state:
+  description: State of the network [Enabled/Disabled].
+  returned: success
+  type: str
+  sample: Enabled
+broadcast_domain_range:
+  description: broadcastdomainrange of the network [POD / ZONE].
+  returned: success
+  type: str
+  sample: ZONE
+isolation_method:
+  description: isolationmethod of the network [VLAN/GRE/L3].
+  returned: success
+  type: str
+  sample: VLAN
+network_speed:
+  description: networkspeed of the network [1G/10G].
+  returned: success
+  type: str
+  sample: 1G
+zone:
+  description: Name of zone the physical network is in.
+  returned: success
+  type: str
+  sample: ch-gva-2
+domain:
+  description: Name of domain the network is in.
+  returned: success
+  type: str
+  sample: domain1
+'''
+
+from ansible.module_utils.basic import AnsibleModule
+from ansible.module_utils.cloudstack import (
+    AnsibleCloudStack,
+    cs_argument_spec,
+    cs_required_together,
+)
+
+
+class AnsibleCloudStackPhysicalNetwork(AnsibleCloudStack):
+
+    def __init__(self, module):
+        super(AnsibleCloudStackPhysicalNetwork, self).__init__(module)
+        self.returns = {
+            'isolationmethods': 'isolation_method',
+            'broadcastdomainrange': 'broadcast_domain_range',
+            'networkspeed': 'network_speed',
+            'vlan': 'vlan',
+            'tags': 'tags',
+        }
+        self.nsps = []
+        self.vrouters = None
+        self.loadbalancers = None
+
+    def _get_common_args(self):
+        args = {
+            'name': self.module.params.get('physical_network'),
+            'isolationmethods': self.module.params.get('isolation_method'),
+            'broadcastdomainrange': self.module.params.get('broadcast_domain_range'),
+            'networkspeed': self.module.params.get('network_speed'),
+            'tags': self.module.params.get('tags'),
+            'vlan': self.module.params.get('vlan'),
+        }
+        if args['tags'] and len(args['tags']) > 1:
+            self.module.fail_json(msg="Failed: Physical networks can only have one tag")
+
+        state = self.module.params.get('state')
+        if state in ['enabled', 'disabled']:
+            args['state'] = state.capitalize()
+        return args
+
+    def get_physical_network(self, key=None):
```

Comment:
I think it's time to move this method to Cloudstack module utils.

Reply:
@dpassante I agree on it too. I wanted to do it in one of the previous PR's but it got quite messy. I want to put it in there because I've got pending changes in `cs_vlan_ip_range` and a new `cs_traffic_type` module that both need it. Although I would still need to override it in this module - the same case with `get_network` method and `cs_network` module.

You can check the code here: https://github.com/ansible/ansible/pull/54020/files#diff-96da40a9131629a9034804485b0d0808 I'll resubmit it once this PR is merged.

Reply:
Ok, I see.